src/components/application_manager/test/resumption/resume_ctrl_test.cc

```diff
@@ -1203,6 +1229,237 @@ TEST_F(ResumeCtrlTest, GetSavedAppHmiLevel_AskedAppFound_INVALID_ENUM) {
             res_ctrl_->GetSavedAppHmiLevel(kTestPolicyAppId_, kMacAddress_));
 }
 
+TEST_F(
+    ResumeCtrlTest,
+    ResumptionLowVoltage_AppInFullUnregisteredWithinTimeFrame_HMILevelRestored) {
+  const mobile_apis::HMILevel::eType restored_test_type = eType::HMI_FULL;
+  const uint32_t time_offset = 5;
+  const uint32_t time_stamp =
+      time(nullptr) - resumption_delay_before_ign_ + time_offset;
+  smart_objects::SmartObject saved_app;
+  saved_app[application_manager::strings::hmi_level] = restored_test_type;
+  saved_app[application_manager::strings::time_stamp] = time_stamp;
+
+  application_manager::CommandsMap command;
+  DataAccessor<application_manager::CommandsMap> data_accessor(
+      command, app_set_lock_ptr_);
+
+  EXPECT_CALL(mock_state_controller_, SetRegularState(_, restored_test_type))
+      .Times(AtLeast(1));
+  GetInfoFromApp();
+  EXPECT_CALL(mock_app_mngr_, GetDefaultHmiLevel(const_app_))
+      .WillRepeatedly(Return(kDefaultTestLevel_));
+  EXPECT_CALL(*mock_app_, commands_map()).WillRepeatedly(Return(data_accessor));
+  ON_CALL(*mock_storage_,
+          GetSavedApplication(kTestPolicyAppId_, kMacAddress_, _))
+      .WillByDefault(DoAll(SetArgReferee<2>(saved_app), Return(true)));
+
+  EXPECT_CALL(*mock_storage_,
+              RemoveApplicationFromSaved(kTestPolicyAppId_, kMacAddress_))
+      .WillOnce(Return(true));
+
+  ON_CALL(mock_app_mngr_, GetUserConsentForDevice("12345"))
+      .WillByDefault(Return(policy::kDeviceAllowed));
+
+  NiceMock<MockApplicationManagerSettings> app_mngr_settings_;
+  EXPECT_CALL(mock_app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+
+  EXPECT_CALL(app_mngr_settings_, resumption_delay_before_ign())
+      .WillOnce(Return(resumption_delay_before_ign_));
+
+  res_ctrl_->SaveLowVoltageTime();
+  res_ctrl_->StartAppHmiStateResumption(mock_app_);
+}
+
+TEST_F(
+    ResumeCtrlTest,
+    ResumptionLowVoltage_AppInFullUnregisteredOnTheBorderOfTimeFrame_HMILevelRestored) {
+  const mobile_apis::HMILevel::eType restored_test_type = eType::HMI_FULL;
+  const uint32_t time_offset = 5;
+  const uint32_t time_stamp =
+      time(nullptr) - resumption_delay_before_ign_ + time_offset;
```

Comment:
@jacobkeeler  On the border it means that app was unregistered in the frame of 30 seconds before low voltage and application should restore it's HMI level.  So there should be `+ time_offset`. 


Here is the example: `[----|-----u-----L]`

 - `|` - 30 seconds frame before the low voltage
 - `u` - unregistering application
 - `L` - low voltage event


In case if there will be `- time_offset` there will be the following picture : 
`[---u---|----L]`, and Application will not restore it's HMI level.

Reply:
I understood "on the border" as "exactly 30 seconds before LOW_VOLTAGE".
Another thing to note here is that the `time_offset` is added to the delay in this particular test case, so the window is `35` seconds.
```
  EXPECT_CALL(app_mngr_settings_, resumption_delay_before_ign())
      .WillOnce(Return(resumption_delay_before_ign_ + time_offset));
```

If this is just testing when an app unregisters **within** the time frame, then this case is redundant, as that case is covered by `ResumptionLowVoltage_AppInFullUnregisteredWithinTimeFrame_HMILevelRestored `

Reply:
@jacobkeeler  fixed 33e6dd415c

Reply:
@jacobkeeler removed 5224964d04